src/bin/inx-chronicle/main.rs

```diff
@@ -107,17 +105,18 @@ async fn main() -> eyre::Result<()> {
     #[cfg(feature = "api")]
     if config.api.enabled {
         use futures::FutureExt;
+        let worker = api::ApiWorker::new(db.clone(), config.api.clone())?;
         let mut handle = shutdown_signal.subscribe();
         tasks.spawn(async move {
-            let worker = api::ApiWorker::new(&db, &config.api)?;
             worker.run(handle.recv().then(|_| async {})).await?;
             Ok(())
         });
     }
 
     // We wait for either the interrupt signal to arrive or for a component of our system to signal a shutdown.
     tokio::select! {
-        _ = process::interrupt_or_terminate() => {
+        res = process::interrupt_or_terminate() => {
+            res?;
```

Comment:
Let's not do this

Reply:
Don't let me always ask why and come forward with an actual explanation please other than I don't like. You know I can be reasoned with, but I would never give you a comment like this without also explaining what I don't like about it.